### DISCUSSION TITLE
Move to/Support another Docker registry other than Docker Hub

### DISCUSSION THREAD

**Discussion by jacroe** [navidrome/navidrome#2257] [Ideas]
Based on the issues described in [this blog post](https://blog.alexellis.io/docker-is-deleting-open-source-images/), Docker Hub is, perhaps, no longer a safe provider of Docker images. Unless the project is willing to pay.

What I would like is for the project to support a different Docker registry, in combination or not, with [GHCR.io](https://ghcr.io/) or [Quay](https://quay.io/).

Quite a bit of additional context can be seen in [Hacker News](https://news.ycombinator.com/item?id=35166317) and this [/r/selfhosted post](https://www.reddit.com/r/selfhosted/comments/11ryub1/fyi_docker_is_deleting_open_source_organisations/).

**Comment by onedr0p** [comment]
Another ref: https://github.com/docker/hub-feedback/issues/2314

**Comment by deluan** [comment]
Yes, I've been thinking about this, and when I received the email I decided to start publishing the image in more than one place, probably will continue with Docker Hub (as I use my personal account, not the Free organization one) and ghcr.io.

**Comment by deluan** [comment]
Seems that Docker chickened-out:
> On March 14, 2023, we emailed you about your Free Team subscription, outlining our intention to sunset that plan. After listening to the concerns of the community, we’ve decided to reverse course, and are no longer sunsetting the Free Team plan.

>If you’re currently on the Free Team plan, you no longer have to migrate to another plan by April 14. All customers who upgraded to a paid subscription will automatically receive a full refund for the transaction in the next 30 days, allowing them to use their new paid subscription for free for the duration of the term they purchased.

> We apologize for the alarm our decision caused. For more details, please [visit our FAQ](https://em.docker.com/NzkwLVNTQi0zNzUAAAGKtH1UE1p5IB1hNVPKNTIhN-iab6BK9_vAcHUlQ-WeHMl84rBkjGCJNxVhByLa3TxnpZ-yXA4=).

**Reply by onedr0p** [reply]
I still wouldn't trust them to keep their word on that. GHCR is so easily to push to if using Github workflows already.

**Reply by deluan** [reply]
Yeah, I know. It didn't change anything, I'm still planning to start a mirror in GHCR.

**Reply by BadCo-NZ** [reply]
> Yeah, I know. It didn't change anything, I'm still planning to start a mirror in GHCR.

Another option is to do what Linuxserver.io does and use their own domain (lscr.io) which you can point at any container repository. Very future proof incase github does the same.

**Comment by deluan** [comment]
Ok, now all images (`develop`, releases and PRs) are published to ghcr.io, and docker hub only receives `develop` and releases.

@onedr0p , what is missing in the original Navidrome image that prompted you to create your own?

**Reply by onedr0p** [reply]
At the time it was only on dockerhub. I absolutely loathe Docker Inc. and Mirantis. I'll give the official one a go again and probably deprecate my image for navidrome.

**Reply by jacroe** [reply]
@deluan Is it possible to get an DockerHub-equivalent `latest` tag up there or do we need to wait until the next release? I've only started digging into GitHub Actions so I don't know if it's possible to do an arbitrary "build" for that or not.

**Reply by onedr0p** [reply]
@deluan can you also copy the released dockerhub images over to GHCR?

https://github.com/google/go-containerregistry/blob/main/cmd/crane/doc/crane_copy.md

Ref: https://github.com/docker/hub-feedback/issues/2314#issuecomment-1470965880

**Reply by deluan** [reply]
@onedr0p Do we really do this? I mean, I rather have users use the latest version, and I'm ok with the old stuff only being available in docker hub, as it probably should not be used after next release.

What is the reasoning for that?

@jacroe Based on the current setup, I can only do this manually, or wait for the next release.

**Reply by onedr0p** [reply]
There's no other reason other than there is no current release on GHCR :)

**Reply by deluan** [reply]
Ok, `latest` and `0.49.3` are there.

**Comment by jacroe** [comment]
@deluan thanks so much for how quickly you were able to support this! I've finally gotten around to switching my `image` location to GHCR everything looks good (as expected). I'll close this as resolved.

Again, thank you, for not only this particular solution but for the entire project. If I don't use it daily, I at least use it weekly. And there's not a lot of any software I can say that about.
